File: src/main/kotlin/me/fungames/jfortniteparse/fort/exports/AthenaEmojiItemDefinition.java
package me.fungames.jfortniteparse.fort.exports;

import me.fungames.jfortniteparse.ue4.objects.core.math.FIntPoint;
import me.fungames.jfortniteparse.ue4.objects.core.math.FLinearColor;
import me.fungames.jfortniteparse.ue4.objects.core.math.FVector;
import me.fungames.jfortniteparse.ue4.objects.gameplaytags.FGameplayTag;
import me.fungames.jfortniteparse.ue4.objects.uobject.FPackageIndex;
import me.fungames.jfortniteparse.ue4.objects.uobject.FSoftObjectPath;

public class AthenaEmojiItemDefinition extends AthenaDanceItemDefinition {
    public FGameplayTag EmojiGameplayCueTag;
    public FSoftObjectPath PreviewAnimation;
    public FSoftObjectPath SpriteSheet;
    public FIntPoint SheetDimensions;
    public Integer FrameIndex;
    public Integer FrameCount;
    public FPackageIndex /*UMaterialInterface*/ BaseMaterial;
    public FVector IconSize;
    public FLinearColor InitialColor;
    public FVector InitialLocation;
    public FVector InitialVelocity;
    public Float LifetimeIntroSeconds;
    public Float LifetimeMidSeconds;
    public Float LifetimeOutroSeconds;
    public FPackageIndex /*UMaterialInstance*/ GeneratedMaterial;
}

File: src/main/kotlin/me/fungames/jfortniteparse/fort/exports/AthenaEmojiItemDefinition.kt
package me.fungames.jfortniteparse.fort.exports

import me.fungames.jfortniteparse.ue4.objects.core.math.FLinearColor
import me.fungames.jfortniteparse.ue4.objects.uobject.FPackageIndex
import me.fungames.jfortniteparse.ue4.objects.uobject.FSoftObjectPath

@ExperimentalUnsignedTypes
class AthenaEmojiItemDefinition : FortItemDefinition() {
    var PreviewAnimation: FSoftObjectPath? = null
    var SpriteSheet: FSoftObjectPath? = null
    var BaseMaterial: FPackageIndex? = null
    var InitialColor: FLinearColor? = null
    var Animation: FPackageIndex? = null
}